### DISCUSSION TITLE
When is method schema_status(const storage_impl<index_t<Cols...>, Tss...>&, sqlite3*, bool) called?

### DISCUSSION THREAD

**Discussion by juandent** [fnc12/sqlite_orm#1015] [General]
Hi

There is a method in class storage_t<...> (see storage.h) that is not invoked by any of the unit-tests. Its signature is:

```C++
template<class... Tss, class... Cols>
            sync_schema_result schema_status(const storage_impl<index_t<Cols...>, Tss...>&, sqlite3*, bool) {
                return sync_schema_result::already_in_sync;
            }
``` 

I need to find when this is called to make sure the corrections to schema_status() are complete and correct. (see PR #1010)

Help will be greatly appreciated....

Juan

**Comment by fnc12** [comment]
it is called when you call `sync_schema` a storage with an index(es)

**Comment by juandent** [comment]
any particular unit-test that exercises this method?
I need to test it to make sure everything is ok

**Comment by fnc12** [comment]
dunno. The amount of unit-tests is so huge that I cannot remember all of them. Try to search using `make_index` inside unit-tests code or create your own. You can copy and paste code from `index.cpp` example

**Comment by juandent** [comment]
the function  `schema_status(const storage_impl<index_t<Cols...>, Tss...>&, sqlite3*, bool)`

is not called by this code:

```C++
    struct User {
        int id = 0;
        std::string name;
    };
    auto storage = make_storage("", make_index("user_name", &User::name),
        make_table("users",
            make_column("id", &User::id, primary_key(), autoincrement()),
            make_column("name", &User::name)));
    storage.sync_schema(true);
```

how can I test that function?

**Comment by fnc12** [comment]
try to comment it out and compile. Does it compile? Also try to call `sync_schema_simulate`

**Comment by juandent** [comment]
Will try! 
Thanks

**Comment by juandent** [comment]
```C++`
storage.sync_schema_simulate(true);
```

succeeds in calling the function!

Thanks Eugene!
